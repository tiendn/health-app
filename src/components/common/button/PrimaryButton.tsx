import React from "react";
import styles from "./style.module.scss";
import { Colors } from "@/theme/colors";

type Props = {
	onClick: () => void;
	text: string;
};

const PrimaryButton: React.FC<Props> = ({ onClick, text }) => {
	return (
		<button
			onClick={onClick}
			className={styles.btnPrimary}
			style={{ backgroundColor: Colors.primary[400] }}
		>
			<span className={styles.btnPrimaryText}>{text}</span>
		</button>
	);
};

export default PrimaryButton;
