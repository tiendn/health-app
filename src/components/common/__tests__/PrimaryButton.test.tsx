import React from "react";
import renderer from "react-test-renderer";
import PrimaryButton from "../button/PrimaryButton";

test("Badge renders correctly", () => {
	const tree = renderer.create(<PrimaryButton onClick={() => {}} text="asdfs" />).toJSON();
	expect(tree).toMatchSnapshot();
});
