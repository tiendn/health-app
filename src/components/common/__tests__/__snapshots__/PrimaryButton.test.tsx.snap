// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<button
  className="btnPrimary"
  onClick={[Function]}
  style={
    {
      "backgroundColor": "#FF963C",
    }
  }
>
  <span
    className="btnPrimaryText"
  >
    asdfs
  </span>
</button>
`;
